src/core/qgspointlocator.cpp

```diff
@@ -332,7 +332,16 @@ static QgsPointLocator::MatchList _geometrySegmentsInRect( QgsGeometry *geom, co
   // we need iterator for segments...
 
   QgsPointLocator::MatchList lst;
-  QByteArray wkb( geom->asWkb() );
+
+  QgsGeometry straightGeom = geom->convertToType( QgsWkbTypes::LineGeometry, true );
+  straightGeom.convertToStraightSegment();
+
+  QgsWkbTypes::Type wkbType = straightGeom.wkbType();
+
+  if ( ( straightGeom.type() != QgsWkbTypes::LineGeometry ) && ( !straightGeom.isMultipart() ) && ( wkbType == QgsWkbTypes::LineString25D ) )
```

Comment:
Is this condition possible? The first check is that it's not a line type, and then the last is that it's a linestring25d?

Reply:
Right. I stupidly copied a piece of code, indeed the linestring25D are only linestringZ, they will be treated like the others. I'm going to remove the last condition.